src/Datasource/EntityTrait.php

```diff
@@ -911,6 +918,16 @@ public function getErrors()
             ->toArray();
     }
 
+    /**
+     * Returns whether this entity contains errors.
+     *
+     * @return bool
+     */
+    public function hasErrors()
+    {
+        return $this->_hasErrors;
```

Comment:
Wouldn’t it be better to just return `!empty($this->_errors)` instead of having a new property?

Reply:
A new property makes the check more strict. It also opens the possibility to set different types of errors without being depended on `$this->_errors`.

Reply:
But wouldn't that make it more confusing? To have hasErrors true, but then reading the error list can be empty etc?
I don't think this is wise.

Reply:
On the other hand having multiple sources of truth also increases the chance of disparity and ending up returning wrong result.

Reply:
Maybe I should rename the property and method to `canBePersisted`. 

Not having errors doesn't mean we the Entity can be persisted. Perhaps this is determined by other variables like, we only want to be sure an Entity has been validated. This also makes more sense regarding the `newEntity()` issue.

Reply:
That might be a semantically more clear approach, yes.